server/src/main/java/com/thoughtworks/go/server/service/BackupService.java

```diff
@@ -135,6 +159,49 @@ public void run() throws IOException {
         }
     }
 
+    private ServerBackup performBackupWithoutAuthentication(Username username,
+                                                            HttpLocalizedOperationResult result,
+                                                            BackupInitiator initiatedBy) {
+        ServerBackup serverBackup = doPerformBackup(username, result);
+
+        String postBackupScriptFile = postBackupScriptFile();
+
+        if (isNotBlank(postBackupScriptFile)) {
+            PostBackupScript postBackupScript = new PostBackupScript(postBackupScriptFile, initiatedBy, username, serverBackup, backupLocation());
+            if (postBackupScript.execute()) {
+                // only set message, retain the original status
+                result.setMessage(joinSentences(result.message(), "Post backup script executed successfully."));
+            } else {
+                result.internalServerError(joinSentences(result.message(), "Post backup script exited with an error, check the server log for details."));
```

Comment:
Just concatenate the messages and set appropriate status code on the `HttpLocalizedOperationResult` object.